src/actions/TemperatureSyncActions.js

```diff
@@ -49,6 +55,39 @@ const scanForSensors = () => async dispatch => {
   if (success) dispatch(scanComplete());
 };
 
+const downloadLogsError = () => ({ type: TEMPERATURE_SYNC_ACTIONS.DOWNLOAD_LOGS_ERROR });
+const downloadLogsStart = () => ({ type: TEMPERATURE_SYNC_ACTIONS.DOWNLOAD_LOGS_START });
+const downloadLogsComplete = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.DOWNLOAD_LOGS_COMPLETE,
+});
+
+const downloadLogs = sensor => async dispatch => {
+  dispatch(downloadLogsStart());
+
+  const downloadedLogsResult = (await sensor?.downloadLogs()) ?? {};
+  const { success = false, data = [] } = downloadedLogsResult;
+
+  if (success && data?.[0]?.logs) {
```

Comment:
Just a thought. I recall seeing a complaint about using `reject` so possibly that part shouldn't be here.

```suggestion
  if (!success || !data?.[0]?.logs) {
    dispatch(downloadLogsError());
    return Promise.reject();
  }
```

and then un-indent the rest of the method